base/Cargo.toml

```diff
@@ -6,7 +6,9 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-lean-sys = { git = "https://github.com/digama0/lean-sys.git", version = "0.0.3" }
+lean-sys = { git = "https://github.com/digama0/lean-sys.git", version = "0.0.3", default-features = false }
 
 [features]
 nightly = []
+small_allocator = ["lean-sys/small_allocator"]
+default = ["small_allocator"]
```

Comment:
By the way, I'm not even sure it's appropriate to have a feature for small_allocator support. I think a constant in the code would be sufficient. The nightly lean which is shipped either has it enabled or not, it's not like the user of lean-sys gets to choose. (It is enabled on nightly.)

Reply:
the problem here is that if user choose to enable it or not which disagrees the lean distribution they use, they will either get a linkage error or a runtime segmentation violation. 

Ideally, lean can provide an option that dumps the feature flags (such as llvm, small_allocator), then we can either set it up in our build stage and still leave it as a feature/cfg but user can decide it properly.

For example, in the integration of lake, if lake knows small_allocator is enabled, it can generates —feature/—cfg small_allocator and pass it to cargo.

Reply:
I think this crate should (explicitly or implicitly) depend on a specific version of lean, because this interface is only as stable as lean itself. However because the interface doesn't actually change that often, it is desirable for it not to fail if the user is using a slightly different version of the toolchain as long as it is close enough to our expectations. The current version of lean uses a small allocator, if they decide to turn it off later then we should do so as well in the update.